src/diagnostics/writers.jl

```diff
@@ -0,0 +1,161 @@
+# Writers.jl
+#
+# This file defines function-generating functions for output_writers for diagnostics. The
+# writers come with opinionated defaults.
+
+import ClimaCore.Remapping: interpolate_array
+import NCDatasets
+
+"""
+    HDF5Writer()
+
+
+Save a `ScheduledDiagnostic` to a HDF5 file inside the `output_dir` of the simulation.
+
+
+TODO: This is a very barebone HDF5Writer. Do not consider this implementation as the "final
+word".
+
+We need to implement the following features/options:
+- Toggle for write new files/append
+- Checks for existing files
+- Check for new subfolders that have to be created
+- More meaningful naming conventions (keeping in mind that we can have multiple variables
+  with different reductions)
+- All variables in one file/each variable in its own file
+- All timesteps in one file/each timestep in its own file
+- Writing the correct attributes
+- Overriding simulation.output_dir (e.g., if the path starts with /)
+- ...more features/options
+
+"""
+
+function HDF5Writer()
```

Comment:
I know it's in a different namespace, but it's a bit confusing that this returns a function rather than a `ClimaCore.InputOutput.HDF5Writer` object.

Reply:
What if instead of passing `ClimaAtmos.Diagnostics.HDF5Writer()`, you instead passed `ClimaCore.InputOutput.HDF5Writer` (i.e. the type), and then we dispatched on that?

Reply:
We can do that, but wouldn't we have to prepare a `ClimaCore.InputOutput.HDF5Writer` for each file we want to write?

In general, I agree that the writer interface can be significantly improved